project/GenSodiumPlugin.scala

```diff
@@ -13,12 +13,23 @@ object GenSodiumPlugin extends sbt.AutoPlugin {
 
   override def buildSettings = Seq(
     /* See plugins.sbt for why this is dynamically loaded */
-    gensodium := {
-      getClass.getClassLoader
-        .loadClass("tsec.build.GenSodium")
-        .getMethod("main", classOf[Array[String]])
-        .invoke(null, Array.empty[String])
+    gensodium := Def.task {
+      if (canWeUseComSunTools_?) {
+        getClass.getClassLoader
+          .loadClass("tsec.build.GenSodium")
+          .getMethod("main", classOf[Array[String]])
+          .invoke(null, Array.empty[String])
+      } else {
+        sys.error("sodium \"macro\" not supported on your superior JDK, sorry")
```

Comment:
We should report/ give a hint what vendor is possible

Reply:
🏳️ Think our sys.error should likely not need a sarcasm flag.

Reply:
Yeah, I get more sarcastic as the night goes on.